core/domain/platform_feature_services.py

```diff
@@ -352,11 +359,17 @@ def get_platform_parameter_schema(param_name: str) -> Dict[str, str]:
     Returns:
         Dict[str, str]. The schema of the platform parameter according
         to the data_type.
+
+    Raises:
+        Exception. The platform parameter does not have valid data type.
     """
     parameter = registry.Registry.get_platform_parameter(param_name)
-    if parameter.data_type == 'string':
-        return {'type': 'unicode'}
-    elif parameter.data_type == 'number':
-        return {'type': 'float'}
+    if DATA_TYPE_TO_SCHEMA.get(parameter.data_type) is not None:
+        schema = copy.deepcopy(DATA_TYPE_TO_SCHEMA[parameter.data_type])
+        return {'type': schema}
     else:
-        return {'type': parameter.data_type}
+        raise Exception(
+            'The %s platform parameter does not have a valid '
+            'data type, must be one of %s inorder to get schema.' % (
```

Comment:
The %s platform parameter has a data type of %s which is not valid. Please use one of these data types instead: %s.

Reply:
Done